src/main/java/com/dislinkt/post/service/PersonService.java

```diff
@@ -76,6 +77,61 @@ public Boolean canInteractWith(UUID id, UUID receiverId) throws Exception{
         return Boolean.TRUE;
     }
 
+    public List<PersonDTO> getFollowing(UUID id) throws Exception {
+        Person user = findOne(id);
+        if (user == null)
+            throw new Exception("user with given id doesn't exist");
+
+        List<Person> following = user.getFollowing().stream()
+                .filter(person -> person.getPrivacy() == ProfilePrivacy.PUBLIC || person.getFollowing().contains(user))
+                .collect(Collectors.toList());
+
+        return mapper.toDtoList(following);
+    }
+
+    public Boolean followPerson(UUID id, UUID idToFollow) {
+        Person user = findOne(id);
+        Person userToFollow = findOne(idToFollow);
+
+        if (user == null || userToFollow == null)
+            return Boolean.FALSE;
+
+        if (user.getFollowing().contains(userToFollow))
+            return Boolean.FALSE;
+
+        if (user.getBlocked().contains(userToFollow) || userToFollow.getBlocked().contains(user))
+            return Boolean.FALSE;
+
+        if (userToFollow.getPrivacy() == ProfilePrivacy.PUBLIC) {
+            user.getFollowers().add(userToFollow);
+        }
+
+        user.getFollowing().add(userToFollow);
+        repository.save(user);
+
+        return Boolean.TRUE;
+    }
+
+    public Boolean unfollowPerson(UUID id, UUID idToUnfollow) {
+        Person user = findOne(id);
+        Person userToFollow = findOne(idToUnfollow);
+
+        if (user == null || userToFollow == null)
+            return Boolean.FALSE;
+
+        if (!user.getFollowing().contains(userToFollow))
+            return Boolean.FALSE;
+
+        if (user.getBlocked().contains(userToFollow) || userToFollow.getBlocked().contains(user))
+            return Boolean.FALSE;
+
+        user.getFollowers().remove(userToFollow);
+        user.getFollowing().remove(userToFollow);
```

Comment:
Similarly to my comment above, I don't think unfollowing a user means he/she has to automatically unfollow you.